core/src/main/java/org/infinispan/util/logging/annotation/Logged.java

```diff
@@ -0,0 +1,24 @@
+package org.infinispan.util.logging.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation should be used on methods that need to be notified when information is logged by the
+ * {@link org.infinispan.util.logging.events.EventLogger}. There is no distinction between the log level or category.
+ * <p/>
+ * Methods annotated with this annotation should accept a single parameter, a {@link
+ * org.infinispan.notifications.cachemanagerlistener.event.LoggedEvent}, otherwise a {@link
+ * org.infinispan.notifications.IncorrectListenerException} will be thrown when registering your listener.
+ * <p/>
+ * Any exceptions thrown by the listener will abort the call. Any other listeners not yet called will not be called.
+ *
+ * @see org.infinispan.notifications.Listener
+ * @since 14.0
+ */
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Logged {
```

Comment:
I am having a hard time with this one, sorry :anguished: The event the notifier publishes is a `LoggedEvent`, an annotation with `@EventLogged` and the argument as `LoggedEvent` could cause some confusion?

I've use the approach as the other annotations to be `@Fooed` and argument `FooedEvent`, but only `Logged` does not seem that meaningful name. If you think is ok the naming for the annotation + argument to be similar I'll update.

Reply:
My only issue with `Logged` on it's own is that it could cause some confusion with normal logging, but maybe that's just in my head :slightly_smiling_face:. I take your point that you're currently following an established convention, so maybe we should just keep it as `@Logged` after all.

Reply:
I'm not really enthusiastic about the introduction of this annotation, as eventually I'd like us to move away from runtime annotation usage (the new API doesn't use annotations for events anymore).

Reply:
I'll apply the same approach of the new API here and remove all the annotation related stuff